### DISCUSSION TITLE
gcTime -> garbageCollectTime

### DISCUSSION THREAD

**Discussion by nicomatsmart** [TanStack/query#8119] [General] [answered]
Hi everyone,

I was reviewing a PR using `gcTime` and had to look up the docs, since the abbreviation wasn't immediately obvious to me. I found the discussion where [cacheTime was renamed to gcTime](https://github.com/TanStack/query/issues/4678)  [(PR)](https://github.com/TanStack/query/pull/4829), and I understand the goal of reducing confusion with `staleTime`. However, I’m wondering if part of the intention behind `gc` was to encourage users to consult the docs if they’re unfamiliar with it? That gc is not exactly garbage collection, and the exact definition should be looked up?

While `gc` is commonly understood in certain circles, something like `garbageCollectTime` would still be more clear.

Thoughts?

**Comment by TkDodo** [comment, marked as answer]
we had a lot of bike-shedding discussions in the v5 roadmap:
- https://github.com/TanStack/query/discussions/4252

I’m usually against abbreviations but this is a compromise for not being too verbose.

It’s also a very well known abbreviation for developers. There’s `git gc` for example. If you’re just starting out with development or react-query, then don’t worry about this setting - you likely don’t need to customize it :)

**Reply by nicomatsmart** [reply]
thanks for the quick answer. I understand the compromise, won't fight it then.
